deploy/crds/infra.watch_servicetelemetrys_crd.yaml

```diff
@@ -167,49 +167,71 @@ spec:
                     description: Events related backend configuration.
                     properties:
                       elasticsearch:
-                        description: Events storage backend ElasticSearch
+                        description: Events storage backend Elasticsearch
                         properties:
                           enabled:
-                            description: Enable ElasticSearch as a storage backend for events
+                            description: Enable Elasticsearch as a storage backend for events
                             type: boolean
+                          forwarding:
+                            description: Configuration for where to forward events
+                            type: object
+                            properties:
+                              hostUrl:
```

Comment:
Note: following results due to code walk through, so some questions and review path happens below, and I think I answer most of my own questions. I think commenting inline at each of the paths would have been more confusing.

~~~

Could the `hostUrl` just be the Service name here and resolve internally to the existing managed ES? If you had a previously deployed ES and removed the `version`, `storage`, etc fields, and inserted the Service name (or one of the resolved-locally DNS names?) could you do an in-place migration to orphan the existing ES from management?

That was one of the items I wanted to test and look for in the code base as I'm not exactly sure what will trigger an `absent` value (looks)... so just setting enabled: true will create the ES manifest: https://github.com/infrawatch/service-telemetry-operator/blob/9eacaf4000b9bf20d21f87ebe48e3ba7613eb2ea/roles/servicetelemetry/tasks/component_elasticsearch.yml#L12

OK, so the creation and other items only happens when strategy is `use_redhat`. OK that makes sense. If you change to `use_redhat`, I don't see anything that will trigger the `state: absent`, so ES will remain it seems.

In this section for grafana (https://github.com/infrawatch/service-telemetry-operator/pull/439/files#diff-720dce1475e3129001153c472f39b8f203f25ebfff3e61cc92c503ba4f5a854cL64-L77) I think it might make sense to put this into a block to still exist for `use_community, use_hybrid`?

I think it might make sense to have the hostUrl (https://github.com/infrawatch/service-telemetry-operator/pull/439/files#diff-468bf2d79dac29401233b72f3937e9f68cb9e7049af7bec4d3969c8eb69affe3R48) default to https://elasticsearch-es-http:9200 which I noticed in the `defaults/main.yml` you set to, so that makes sense. You have a couple other example spots, and it's probably worth to just set the value to that everywhere.

https://github.com/infrawatch/service-telemetry-operator/pull/439/files#diff-10e9e317481eb8f22725d09cf3555482d690ff87e2b60861ea51467669b4da9fR32 would then get the proper value if the CR was created with defaults everywhere.

Only last thing I think to look at is https://github.com/infrawatch/service-telemetry-operator/pull/439/files#diff-b661f3e2be8a3856711ae0ed99adda16e30a66a972ec5dc5118f57bfbf3ffe2dR16-R25 which uses user.key and user.crt instead of the former tls.crt and tls.key. Does the certificates generated by our existing deployment contain those fields and would their existing values work against the pre-deployed system?

Now that I've walked through this I want to do a quick deploy and switch from use_community to use_redhat. If it doesn't work, it looks really close to working.

Reply:
Last thought here. There could potentially be a KCS for anyone who wanted to stand up an ES in a similar format to how we were doing it via managed. That gets us off the hook for managing, but still gives a way to get the system operational how it was before, and removing from product docs, but still have a sort of example configuration that the defaults work against. Possibly reduces some maintenance burden and gives someone a place to start, or at least easier for us to build into CI.

Reply:
"I think commenting inline at each of the paths would have been more confusing." Oh yeah? You won't mind if I meatloaf my responses, then, too, right?

> Could the hostUrl just be the Service name here and resolve internally to the existing managed ES?

Yes

> In this section for grafana ...  a block to still exist for use_community, use_hybrid?

I'm not sure what your suggesting exactly. Currently the grafana tasks are only included `in ['use_community', 'use_hybrid']` https://github.com/infrawatch/service-telemetry-operator/blob/master/roles/servicetelemetry/tasks/main.yml#L101  and the block you call out only run if there is an elasticsearch to configure. https://github.com/infrawatch/service-telemetry-operator/pull/439/files#diff-720dce1475e3129001153c472f39b8f203f25ebfff3e61cc92c503ba4f5a854cL64 Are you suggesting something else?

> hostUrl default to https://elasticsearch-es-http:9200/ [but CRD defaults it to https://external-elastic-http.domain:9200"]

Maybe I need to take a second look at this in the context of the "in place migration" like you mention. The example in the CSV is to emphasize that it is an EXTERNAL ES. If you are creating a fresh STF instance with default values, that internal service name is not going to be what you want, either, you absolutely must fill it in. I will test what happens when you upgrade from an old STF version - I think you might be right that the migration will be smoother if the default value matches the legacy value.

> Does the certificates generated by our existing deployment contain those fields and would their existing values work against the pre-deployed system?

No and no. However the dummy cert/key that is included when these fields are missing takes care of that. To clarify - our current setup does not rely on certificate authentication, but the SG has it hard-coded as being required. Previously it supplied the cert/key used by ES itself, which is nonsensical. So now we just supply the dummy cert instead.

> There could potentially be a KCS for anyone who wanted to stand up an ES in a similar format to how we were doing it via managed

Sure. I've added a note to my ticket about CI work, since I'll have all the artifacts in front of me when I do that.

Bottom line:

- I will test in-place migration from use_hybrid (w/ ES) -> use_redhat (w/ existing but now unmanaged ES) 
- A KCS article will cover how to set up a community ES and configure STF to point at it

Reply:
> "I think commenting inline at each of the paths would have been more confusing." Oh yeah? You won't mind if I meatloaf my responses, then, too, right?

Not at all.

> > Could the hostUrl just be the Service name here and resolve internally to the existing managed ES?
> 
> Yes

✅ 

> 
> > In this section for grafana ...  a block to still exist for use_community, use_hybrid?
> 
> I'm not sure what your suggesting exactly. Currently the grafana tasks are only included `in ['use_community', 'use_hybrid']` https://github.com/infrawatch/service-telemetry-operator/blob/master/roles/servicetelemetry/tasks/main.yml#L101 and the block you call out only run if there is an elasticsearch to configure. https://github.com/infrawatch/service-telemetry-operator/pull/439/files#diff-720dce1475e3129001153c472f39b8f203f25ebfff3e61cc92c503ba4f5a854cL64 Are you suggesting something else?

Ah I missed that part. I know the events datasource is only used in the virtual machine view dashboard for VM create/delete annotations. We disabled the annotations by default, but if someone was using use_community, and then enabled the annotations in that dashboard, would they still work without this block?

> > hostUrl default to https://elasticsearch-es-http:9200/ [but CRD defaults it to https://external-elastic-http.domain:9200"]
> 
> Maybe I need to take a second look at this in the context of the "in place migration" like you mention. The example in the CSV is to emphasize that it is an EXTERNAL ES. If you are creating a fresh STF instance with default values, that internal service name is not going to be what you want, either, you absolutely must fill it in. I will test what happens when you upgrade from an old STF version - I think you might be right that the migration will be smoother if the default value matches the legacy value.

I didn't noticed anything that would necessarily cause an issue, but it seems like having everything match at least makes things more consistent, and if you needed to change the value anyways, I see no reason to make it different.
 
> > Does the certificates generated by our existing deployment contain those fields and would their existing values work against the pre-deployed system?
> 
> No, the fields do not exist in the tls material generated by our existing deployments. However the dummy cert/key that is included when these fields are missing takes care of that. To clarify - our current setup does not rely on certificate authentication, but the SG has it hard-coded as being required. Previously we supplied the cert/key used by ES itself, which is nonsensical. So now we just supply the dummy cert instead.

OK thanks for clarifying that.

> > There could potentially be a KCS for anyone who wanted to stand up an ES in a similar format to how we were doing it via managed
> 
> Sure. I've added a note to my ticket about CI work, since I'll have all the artifacts in front of me when I do that.

✅ 

> Bottom line:
> 
> * I will test in-place migration from use_hybrid (w/ ES) -> use_redhat (w/ existing but now unmanaged ES)
> * A KCS article will cover how to set up a community ES and configure STF to point at it

I can create a separate issue in the backlog for the KCS if you want? I don't think that really fits into any of the existing scoped work.

Reply:
>  if someone was using use_community, and then enabled the annotations in that dashboard, would they still work without this block?

If there is an ES configured, then the block _will_ be included and the annotations will work. I didn't test the annotations themselves, but the "Save and test" in the grafana datasource config worked.

> I see no reason to make it different

Okay, agreed.

> create a separate issue in the backlog

STF-1448